### PR TITLE
Replace library of useActiveWeb3React with wagmi hook

### PR DESCRIPTION
**Pull request by nkrulikovsky**

### CODE REVIEW

File: src/hooks/useAddTokenToMetamask.ts [hunk 1]
```diff
@@ -1,43 +1,36 @@
...
-            //@ts-ignore // need this for incorrect ethers provider type
-            type: "ERC20",
-            options: {
-              address: token.addresses[chainId],
-              symbol: token.symbol,
-              decimals: token.decimals,
-              image: getTokenIconPath(token.symbol),
-            },
-          },
+    if (activeConnector && isMetaMask && token) {
```
Comment (hammeiam) [inline src/hooks/useAddTokenToMetamask.ts:22]: Is this the same as `canAdd`?

File: src/hooks/useContract.ts [hunk 2]
```diff
@@ -141,17 +145,20 @@ export function usePoolRegistry(): PoolRegistry | null {
...
     ) as PoolRegistry
-  }, [contractAddress, library])
+  }, [contractAddress, provider, signer])
 }
 
 export function usePoolRegistryMultiCall(): MulticallContract<PoolRegistry> | null {
-  const { library } = useActiveWeb3React()
+  const { data: signer } = useSigner()
+  const provider = useProvider()
+  const signerOrProvider = signer || provider
```
Comment (hammeiam) [inline src/hooks/useContract.ts:160]: Why not use `signerOrProvider` from `useActiveWeb3React`?

File: src/utils/gauges.ts [hunk 3]
```diff
@@ -630,19 +648,19 @@ function getGaugeRewardTokens(
...
+  registryAddresses?: Partial<Record<string, string>>,
   account?: string,
 ) {
-  const childGaugeFactoryAddress = registryAddresses[CHILD_GAUGE_FACTORY_NAME]
+  const childGaugeFactoryAddress = registryAddresses?.[CHILD_GAUGE_FACTORY_NAME]
 
-  if (!childGaugeFactoryAddress)
-    throw new Error("Unable to retrieve and build gauge data")
+  if (!childGaugeFactoryAddress) return
+  // throw new Error("Unable to retrieve and build gauge data")
```
Comment (hammeiam) [inline src/utils/gauges.ts:660]: Now that you're done debugging, should this error be added back in?

File: src/index.tsx [hunk 4]
```diff
@@ -1,27 +1,21 @@
 import "./index.css"
 import "./i18n"
 
-import * as Sentry from "@sentry/react"
+// import * as Sentry from "@sentry/react"
```
Comment (hammeiam) [inline src/index.tsx:4]: Add sentry back in

File: src/pages/AppContainer.tsx [hunk 5]
```diff
@@ -31,20 +32,22 @@ const AppWrapper = styled("div")(({ theme }) => {
...
-          position="top-left"
-        />
-        <DevTool />
-        <WrongNetworkModal />
-      </AppWrapper>
-    </LocalizationProvider>
+    <>
+      <TopMenu />
+      <Suspense fallback={null}>
+        <LocalizationProvider dateAdapter={AdapterDateFns}>
```
Comment (hammeiam) [inline src/pages/AppContainer.tsx:38]: Why not keep `LocalizationProvider` on the outside?

File: src/pages/Pools/Pools.tsx [hunk 6]
```diff
@@ -72,6 +75,8 @@ function Pools(): ReactElement | null {
     })
   }, [account, chainId])
 
+  if (isConnecting) return <div> Loading ...</div>
```
Comment (hammeiam) [inline src/pages/Pools/Pools.tsx:78]: If you're going to display a loading state, can you create a better component for it than this?

File: src/hooks/useAddTokenToMetamask.ts (hunk 1 above)
Comment (nkrulikovsky) [reply src/hooks/useAddTokenToMetamask.ts:22]: Yes, we can replace it with `canAdd`

File: src/pages/Pools/Pools.tsx (hunk 6 above)
Comment (nkrulikovsky) [reply src/pages/Pools/Pools.tsx:78]: I added this while debugging of `underlying netowork connection`, but these are not related that issue and we don't have much need to displaying loading component while connect account. So, I'll remove this lines.

File: src/hooks/useContract.ts (hunk 2 above)
Comment (nkrulikovsky) [reply src/hooks/useContract.ts:160]: While debugging, I used `signer` and `provider` separately. We can use`signerOrProvider`.

**Pull request merged**